rct229/rulesets/ashrae9012019/section5/section5rule38.py

```diff
@@ -0,0 +1,251 @@
+            else:
+                target_shgc_semiheated = table_G34_lookup(
+                    climate_zone,
+                    "SEMI-EXTERIOR",
+                    "SKYLIGHT",
+                    skylit_wwr=0,
+                )["solar_heat_gain_coefficient"]
+
+            return {
+                "surface_conditioning_category_dict": get_surface_conditioning_category_dict(
+                    climate_zone, building_b
+                ),
+                "target_shgc_res": target_shgc_res,
+                "target_shgc_nonres": target_shgc_nonres,
+                "target_shgc_semiheated": target_shgc_semiheated,
+            }
+
+        def list_filter(self, context_item, data=None):
+            surface_b = context_item.baseline
+            scc = data["surface_conditioning_category_dict"][surface_b["id"]]
+
+            return (
+                get_opaque_surface_type(surface_b) == OST.ROOF
+                and surface_b.get("subsurfaces")
+                and scc != SCC.UNREGULATED
+            )
+
+        class RoofRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section5Rule38.BuildingRule.RoofRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    each_rule=Section5Rule38.BuildingRule.RoofRule.SubsurfaceRule(),
+                    index_rmr="baseline",
+                    list_path="subsurfaces[*]",
+                    manual_check_required_msg=MANUAL_CHECK_MSG,
+                )
+
+            def create_data(self, context, data=None):
+                surface_b = context.baseline
+                surface_id_b = surface_b["id"]
+
+                return {"surface_id_b": surface_id_b}
+
+            class SubsurfaceRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section5Rule38.BuildingRule.RoofRule.SubsurfaceRule, self
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                        required_fields={
+                            "$": ["classification", "glazed_area", "opaque_area"]
+                        },
+                    )
+
+                def is_applicable(self, context, data=None):
+                    subsurface_b = context.baseline
+
+                    return (
+                        subsurface_b.get("classification")
+                        == SURFACE_CLASSIFICATION.DOOR
+                        and subsurface_b.get("glazed_area")
+                        > subsurface_b.get("opaque_area")
+                    ) or subsurface_b.get(
+                        "classification"
+                    ) != SURFACE_CLASSIFICATION.DOOR
+
+                def get_calc_vals(self, context, data=None):
+                    subsurface_b = context.baseline
+                    subsurface_b_shgc = subsurface_b["solar_heat_gain_coefficient"]
```

Comment:
change this variable name to `subsurface_shgc_b` to be consistent with the other variables.